docs/Observability_using_OpenTelemetry.md

```diff
@@ -0,0 +1,897 @@
+# Observability in Data Projects Using OpenTelemetry
+
```

Comment:
Remove the following TODOs to merge the PR:

1. the 1st and 2nd of updating sample code link will be resolved in this PR. #933 
2. the others will **NOT** be included in this PR but be tracked with issues #937 , #936 , #935 , #934